xlive/Blam/Engine/objects/objects.cpp

```diff
@@ -34,45 +60,634 @@ void* object_header_block_get_with_count(const datum object_datum, const object_
+	{
+		return true;
+	}
+	if (TEST_BIT(FLAG(object_header->object_type), _object_type_machine) && object_type_compute_activation(object_index, cluster_activation, &result) || object_header->cluster_index == NONE)
+	{
+		return result;
+	}
+	return (TEST_FLAG(FLAG(object_header->cluster_index & 0x1F), cluster_activation[object_header->cluster_index >> 5].cluster_bitvector));
+}
+
+
+bool set_object_position_if_in_cluster(s_location* location, datum object_index)
+{
+	object_datum* object = object_get_fast_unsafe(object_index);
+	scenario_location_from_point(location, &object->object_origin_point);
+
+	if (location->cluster == NONE)
 	{
-		if (placement_data->tag_index == NONE) 
+		collision_bsp_test_sphere_result test_result;
+		collision_bsp_test_sphere(global_collision_bsp_get(), 0, NULL, &object->object_origin_point, object->shadow_sphere_radius, &test_result);
+
+		if (test_result.stack_depth)
 		{
-			return result;
+			scenario_location_from_leaf(location, test_result.buffer3.leaf_index);
 		}
+		else
+		{
+			scenario_location_from_point(location, &object->position);
+		}
+	}
 
-		object_type_adjust_placement(placement_data);
+	return location->cluster != NONE;
+}
+
+// Gets important info about the object and populates the s_object_payload argument with the appropriate data
+void get_object_payload(datum object_index, s_object_payload* payload)
+{
+	const object_datum* object = object_get_fast_unsafe(object_index);
+	uint16 object_collision_cull_flags = 0;
+	if (object->object_flags.test(_object_has_collision_bit))
+	{
+		object_collision_cull_flags = collision_compute_object_cull_flags(object_index);
 	}
-	*/
-	return INVOKE(0x136CA7, 0x125B77, object_new, placement_data);
+
+	payload->object_type = object->object_identifier.get_type();
+	payload->object_collision_cull_flags = object_collision_cull_flags;
+	payload->origin_point = object->object_origin_point;
+	payload->bounding_sphere_radius = object->shadow_sphere_radius;
+	return;
 }
 
-//Pass datum from new object into object to sync
-void __cdecl simulation_action_object_create(datum object_idx)
+// Reconnects an object to the current bsp that's loaded
+void object_reconnect_to_map(s_location* location, datum object_index)
 {
-	INVOKE(0x1B8D14, 0x1B2C44, simulation_action_object_create, object_idx);
+	s_object_header* object_header = (s_object_header *)datum_get(object_header_data_get(), object_index);
+	object_datum* object = (object_datum *)object_header->object;
+
+	bool cluster_index_is_null = object_header->cluster_index == NONE;
+	s_location* p_location = location;
+
+	if (!p_location)
+	{
+		s_location scnr_location;
+		scenario_location_from_point(&scnr_location, &object->object_origin_point);
+		p_location = &scnr_location;
+		if (scnr_location.cluster == NONE)
+		{
+			scenario_location_from_point(&scnr_location, &object->position);
+		}
+	}
+
+	if (p_location->cluster == NONE)
+	{
+		object->object_flags.set(_object_data_bit_18, true);
```

Comment:
_object_data_bit_18 is _object_data_disconnected_from_map?

Reply:
yea, i'll change that